MapboxNavigation/Error.swift

```diff
@@ -1,19 +1,14 @@
 import Foundation
-
-
-/**
- Custom Error code key used in returned in `voiceController(_:didInterrupt:with:)`.
- */
-public let SpokenInstructionErrorCodeKey = "SpokenInstructionErrorCodeKey"
+import MapboxCoreNavigation
 
 extension NSError {
     /**
      Creates a custom `Error` object.
      */
-    public convenience init(localizedFailureReason: String, detailedFailureReason: String, code: MapboxNavigationError = .defaultError) {
-        self.init(domain: MGLErrorDomain, code: code.rawValue, userInfo: [
+    public convenience init(localizedFailureReason: String, code: MapboxNavigationError = .defaultError) {
+        self.init(domain: MBErrorDomain, code: code.rawValue, userInfo: [
             NSLocalizedFailureReasonErrorKey: localizedFailureReason,
-            SpokenInstructionErrorCodeKey: detailedFailureReason
+            MBSpokenInstructionErrorCode: code
```

Comment:
Sorry for not being clearer in https://github.com/mapbox/mapbox-navigation-ios/pull/800#discussion_r149533375, https://github.com/mapbox/mapbox-navigation-ios/pull/800#discussion_r149804673, and https://github.com/mapbox/mapbox-navigation-ios/pull/800#discussion_r150964273:

* This method should accept a `spokenInstructionCode` parameter of type SpokenInstructionErrorCode.
* SpokenInstructionErrorCode should be an enumeration with the following values: `noAwsResult`, `emptyAwsResponse`, and `audioPlayerFailedToPlay`.
* MBErrorCode (renamed from MapboxNavigationError) should have the following values: `unknown` and `spokenInstructionFailed`.
* One of the call sites in PollyVoiceController should call this initializer passing in `spokenInstructionFailed` as the `code` and `noAwsResult` as the `spokenInstructionCode`.